Arelle/Arelle
# Implement integration test script runner

#### Reason for change
There will be a need for running general integration test scripts, separate from the conformance suite pattern we currently use. These integration tests should be as flexible as possible.

#### Description of change
- Implemented standalone `download_cache.py` script for downloading public caches in a cross platform way (to remove the need separate conditional steps in workflows)
- Created script running framework in `tests/integration_tests/scripts` based off of the existing conformance suite running framework
- Created `requirements-plugins.txt` requirements set for integration tests and builds
- Implemented test for Japan IXDS (Resolves #901)
- Implemented test for triggering `ixbrl-viewer` from the command line (to help test #734)
- Created workflow to run integration test scripts on pull requests
  - Added dispatch-only version that runs the tests in parallel 
- Added steps to build workflows to run these tests against builds

#### Steps to Test
Should pass on this PR (pull_request trigger):
- [ ] test-scripts.yml

Run on fork (see recent runs on my fork [here](https://github.com/aaroncameron-wk/arelle-public/actions)):
- [ ] test-scripts-parallel.yml
- [ ] conformance-suites.yml
- [ ] build-macos.yml (Note: codesign/notarize can now be optionally skipped)
- [ ] build-windows.yml
- [ ] build-linux.yml

**review**:
@Arelle/arelle
